rfcs/experimental/flow_graph_wait_try_put_and_wait/README.md

```diff
@@ -0,0 +1,608 @@
+metainformation is preserved.
+
+The reference counter in the ``wait_context`` inside of the ``message_metainfo`` is increased in the following cases:
+* The task associated with the computations of the ``try_put_and_wait`` input message is created (*).
+* The item associated with the computations of the desired input is stored in the buffering node or in the internal buffer of the other node type.
+* When the ``continue_msg`` with the non-empty associated waiter is received by ``continue_node``. In this case, the node buffers the ``message_metainfo``s received from each
+  predecessor and increases the underlying reference counter to prolonge the wait until signals from each predecessor is received. 
+
+The reference counter is decreased when:
+* The task associated with the computations of the ``try_put_and_wait`` input is finalized. If the output of the task should be propagated to the successors of the node, it is done
+  from the task body and hence the reference counter is decreased only after creating the task for each successor and increasing the reference counter for them.
+* The item associated with the computations is taken from the buffering node or from the internal buffer.
+* When the desired number of signals from the predecessors was received by the ``continue_node``. This case is equivalent to retrieving the set of buffered ``message_metainfo``s received previously
+  from each predecessor. 
+
+``message_metainfo`` class may be reused in the future to support additional use-cases when it is required to push additional information about the input message through the graph. E.g. supporting
+priorities for single messages - the corresponding priority tag can be assigned as part of the metainfo.
+
+Metainformation class supports containing multiple reference counters at the same time to support joining messages with different associated reference counters in the ``join_node``. 
+See [separate section](#details-about-metainformation-class) for more details.
+
+Currently processing the metainformation is exposed by adding new virtual member functions to various Flow Graph instances. 
+Note that the described functions are internal, may change over time and should not be used directly.
+
+``` cpp
+template <typename T>
+class receiver {
+protected:
+    virtual bool try_put_task(const T& t) = 0; // Existing API
+    virtual bool try_put_task(const T& t, const message_metainfo& info) = 0; // New API
+};
+```
+
+For each particular implementation of ``receiver``, the ``try_put_task`` performs ab action that is required by the corresponding Flow Graph node. 
+
+It may buffer both ``t`` and ``metainfo`` or broadcast the result and the ``info`` to the successors of the node. 
+
+The existing API ``try_put_task(const T& t)`` can reuse the new one with the empty metainfo object if an empty metainfo is a preserved as a lightweight structure.
```

Comment:
I would instead say "if that does not incur a noticeable overhead for the normal use of flow graphs."

Reply:
Applied